packages/react/slot/src/Slot.tsx

```diff
@@ -61,9 +61,11 @@ const SlotClone = React.forwardRef<any, SlotCloneProps>((props, forwardedRef) =>
   const { children, ...slotProps } = props;
 
   if (React.isValidElement(children)) {
+    const childrenRef = getElementRef(children);
     return React.cloneElement(children, {
       ...mergeProps(slotProps, children.props),
-      ref: forwardedRef ? composeRefs(forwardedRef, (children as any).ref) : (children as any).ref,
+      // @ts-ignore
+      ref: forwardedRef ? composeRefs(forwardedRef, childrenRef) : childrenRef,
```

Comment:
Setting the `ref` property has always yielded a type error, just didn't break the build before

Reply:
I wonder what's the future of the Slot component as React considers `cloneElement` a `legacy api`. 🤔

Reply:
> I wonder what's the future of the Slot component as React considers cloneElement a legacy api. 🤔

We'll see—I haven't seen a good alternative there, and "legacy" doesn't mean it's deprecated yet. I think it will stay around for a while.